src/core/components/Paginate/index.js

```diff
@@ -0,0 +1,63 @@
+import React, { PropTypes } from 'react';
+
+import './Paginate.scss';
+
+export default class Paginate extends React.Component {
+  static propTypes = {
+    count: PropTypes.number.isRequired,
+    currentPage: PropTypes.number.isRequired,
+    pager: PropTypes.func.isRequired,
+    perPage: PropTypes.number,
+    showPages: PropTypes.number,
+  }
+
+  static defaultProps = {
+    perPage: 25,
+    showPages: 9,
+  }
+
+  pageCount() {
+    return Math.ceil(this.props.count / this.props.perPage);
+  }
+
+  visiblePages() {
+    const { currentPage, showPages } = this.props;
+    const pageCount = this.pageCount();
+    const pageLinkCount = Math.min(showPages, pageCount);
+    // Offset should be a number between 0 and showPages. When near the middle it will be
+    // `showPages / 2`, when near the begging it will be near 0, when near then end it will be near
+    // `showPages`.
+    const offset = Math.max(
+      // Limit the offset when near the start to `currentPage - 1`, otherwise use `showPages / 2`.
+      Math.min(Math.floor(showPages / 2), currentPage - 1),
+      // When near the end the offset should approach `showPages`, but not if we have fewer pages
+      // than `showPages`.
+      showPages < pageCount ? showPages - (pageCount - currentPage) - 1 : 0);
+    return new Array(pageLinkCount).fill(0).map((val, i) => i + currentPage - offset);
```

Comment:
Might be worth a comment here to explain the use of `fill()` to actually create an array with + `pageLinkCount` items.

Also I see `fill` is not supported in IE or Opera [1] so if this gets re-used on AMO we'd need to patch that. It might be nice to polyfill it now instead.

[1] https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/fill#Browser_compatibility